components/automate-ui/scripts/build.sh

```diff
@@ -1,14 +1,16 @@
 #!/bin/bash
-
 # The build machine is using sh; need to use bash to be able to use PIPESTATUS.
 
+CMD_STRING=$1
+WARNING_LABEL=$2
+
 # Setup an unused stream to redirect to stdout in the next command.
 exec 5>&1
-# We want to capture the `ng test` output to reuse it (rather than have to re-run the command),
+# We want to capture the target command's output to reuse it (rather than have to re-run the command),
 # so sending to an unused stream and redirecting that to stdout (stream 1) where it can be captured into the `out` variable.
-# In order to check the exit code of the `ng test` itself, need to use PIPESTATUS
+# In order to check the exit code of the target command itself, need to use PIPESTATUS
 # to grab the exit code of the first pipe component rather than the final one.
-out=$(ng test --watch=false --code-coverage --source-map=false --configuration=no_auth | tee /dev/fd/5; exit "${PIPESTATUS[0]}")
+out=$(eval "$CMD_STRING | tee /dev/fd/5; exit \${PIPESTATUS[0]}")
```

Comment:
💭 Is the `eval` really necessary? What's up with `$CMD_STRING | tee /dev/fd/5; exit ${PIPESTATUS[0]}`?

Reply:
> Is the eval really necessary?

Yes; without it we are just giving the shell a string, not a command.

> What's up with $CMD_STRING | tee /dev/fd/5; exit ${PIPESTATUS[0]}?

That was designed for flexibility, allowing the $CMD_STRING itself to contain pipes. That had actually been needed in the code base a year or more ago when I originally wrote the code. The _current_ consumers are not passing values containing pipes, but I chose to leave the extra support mechanism in place. If you feel strongly about removing because it is not being used at the moment, I could do that, just let me know.

Reply:
Thanks! I generally like to avoid eval. I guess we could put the label first, use `shift`, and do something like `out=$("$@")`.... but this works so 🤷